schema@v1.1.1/schema.json

```diff
@@ -52,7 +52,7 @@
     "idString": {
       "type": "string",
       "minLength": 1,
-      "pattern": "^[a-z][ A-Za-z0-9]*$"
+      "pattern": "^[a-z][ A-Za-z]*[_0-9]*$"
```

Comment:
This regex is more liberal than the textual description. A better version would be (including @lbam suggestion to remove the white space):

*Note:* untested!
```javascript
"pattern": "^[a-z][A-Za-z]*(?<=_)(?:\d+)$"
```
This uses a lookbehind assertion `(?<_)` that matches `(?:\d+)` only if it is preceded by an underscore (exactly as stated by the textual description).

I'm using:
-  a non-capturing group `(?:x)`, 
- the character class `\d`instead of the more verbose `[0-9]` 
- and the quantifier `+` as we don't want id's that end in an underscore, right?

Reply:
BTW the textual descriptions mentions _in de vorm van een letter reeks_, which implies the minimal length of the `idString` is at least two. This is something that the regex doesn't yet capture. Is _reeks_ wrongly worded? If not, the regex needs to be adjusted beyond what I suggested in https://github.com/Amsterdam/amsterdam-schema/pull/323/files#r756609496

Reply:
Why would it be at least two?

Reply:
> Why would it be at least two?

Because according to VanDale:

Betekenis 'reeks'

reeks (de; v(m); meervoud: reeksen) 
1. rij van dingen die op volgorde staan; = serie: een reeks van jaren
2. 2rij getallen die allemaal op dezelfde manier verschillen met het voorgaande

The words _dingEN_ en _getallEN_ imply more than one thing

So either adjust de description as I suggested in this [comment](https://github.com/Amsterdam/amsterdam-schema/pull/323/files#r756613023) or adjust the regex definition in the pattern. Both need to match, they currently don't.

Reply:
Fair enough.

Re: the regexp, I don't think Python has those lookback assertions. We should do something that is portable to at least Python and JavaScript. Third-party JSON Schema validators may use the regexp libs in their implementation languages, too. This should work in Python and JavaScript:


```
^[a-z][A-Za-z]*(?:_[0-9]+)?$
```

It also avoids the `\d` syntax that has to be escaped to even appear in JSON:

```
$ echo '"\d"' | jq .
parse error: Invalid escape at line 1, column 4
```

Reply:
Python [does support them](https://docs.python.org/3/library/re.html#regular-expression-syntax). However your regex is simpler; I like that better.

Oh and BTW didn't realise the need to escape the `\d`, good catch!